src/include/rados.h

```diff
@@ -118,6 +118,7 @@ struct ceph_eversion {
 #define CEPH_OSD_NEARFULL (1<<5) /* osd is at or above nearfull threshold */
 #define CEPH_OSD_BACKFILLFULL (1<<6) /* osd is at or above backfillfull threshold */
 #define CEPH_OSD_DESTROYED (1<<7) /* osd has been destroyed */
+#define CEPH_OSD_DESTROYING (1<<8) /* osd is being destroyed */
```

Comment:
I don't think we should burn a bit here.  This in-between state is an implementation artifact/ugly of the current monitor, not something that should be exposed via the OSDMap.  If we need to record in-progress state let's do it in a separate set of key/value pairs in the db?